### PR TITLE
[feat] Create single topic for new uploads

### PR DESCRIPTION
**Pull request by SteveHNH**
New uploads can be advertised on a single topic and the header can be
used if it is meant for a downstream service. This is meant to simplify
ingress code in the long run. It also allows services to onboard easier
by just sending data into the platform for it to be sorted at the client
level.

Signed-off-by: Stephen Adams <tsadams@gmail.com>

## Secure Coding Practices Checklist Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Checklist
- [ ] Input Validation

- [ ] Output Encoding

- [ ] Authentication and Password Management

- [ ] Session Management

- [ ] Access Control

- [ ] Cryptographic Practices

- [ ] Error Handling and Logging

- [ ] Data Protection

- [ ] Communication Security

- [ ] System Configuration

- [ ] Database Security

- [ ] File Management

- [ ] Memory Management

- [ ] General Coding Practices

### CODE REVIEW

**Review by aleccohan — APPROVED** [review]
Looks great to me!!

**Review by tahmidefaz — APPROVED** [review]
Looks good!

File: internal/queue/queue.go [hunk 1]
```diff
@@ -4,6 +4,7 @@ import (
 	"time"
 
 	l "github.com/redhatinsights/insights-ingress-go/internal/logger"
+	val "github.com/redhatinsights/insights-ingress-go/internal/validators"
```
Comment (dehort) [inline internal/queue/queue.go:7]: Nitpick...feel free to ignore...to me, "validators" is more clear that "val".

File: internal/validators/types.go [hunk 2]
```diff
@@ -35,6 +37,11 @@ type Metadata struct {
 	StaleTimestamp time.Time `json:"stale_timestamp"`
 }
 
+type Transport struct {
```
Comment (dehort) [inline internal/validators/types.go:40]: Nitpick, I'm not sure Transport is a good name here.  "Transport" makes me think of a socket or an connection of some sort.  Maybe something like "ValidationEvent", "ValidationMessage".  I'm not sure.  Naming things is hard.  :)

File: internal/validators/kafka/kafka.go [hunk 3]
```diff
@@ -97,11 +101,19 @@ func (kv *Validator) Validate(vr *validators.Request) {
...
-	kv.ValidationProducerMapping[realizedTopicName] <- data
+	message := validators.Transport{
+		Message: data,
+		Headers: []kafka.Header{{
+			Key: "service",
+			Value: []byte(vr.Service),
+		}},
+	}
+	kv.ValidationProducerMapping[realizedTopicName] <- message
+	kv.ValidationProducerMapping[config.Get().AnnounceTopic] <- message
```
[the remark this answers is not in the document]
Comment (SteveHNH) [reply internal/validators/kafka/kafka.go:112]: Yeah, I need to handle the backward compatibility

**Comment by lphiri** [comment]
@SteveHNH @jhjaggars - just to be clear, if a new app wants to have a separate announce channel, that will be supported correct?  I think the case of an app wanting a separate secure topic is a real a use case, especially once we move to Managed kafka and start supporting doing things across clusters.

**Comment by SteveHNH** [comment]
@lphiri I don't see why we couldn't pull that off. We could move to more of a "produce to both" logic to more of an if/else thing. If we have a specific topic configured, produce there only. Otherwise send to `announce`. I might not be thinking of the whole picture here though.

**Comment by jhjaggars** [comment]
@lphiri Are you saying that a single announce topic isn't secure enough for uses cases we already have?

**Comment by lphiri** [comment]
@jhjaggars  I was thinking more about future usecases, but its possible some of the data apps are sending today my be considered sensitive.

**Comment by jhjaggars** [comment]
I was thinking about the following path:
1. introduce singular global topic for everything that exists today, and leave existing topics in place
2. allow time to migrate to single topic, deleting single use topics as we can
_if_ we encounter a specific ask then:
3. preserve single-type topics as necessary

**Comment by lphiri** [comment]
@jhjaggars that works - i was essentially asking we have the option to do 3, if needed.

**Review by dehort — APPROVED** [review]
Looks good to me.  I left one suggestion though.  :)

**Pull request merged**